datastore/mongodb/filter_dimensions.go

```diff
@@ -286,7 +286,19 @@ func (c *Client) RemoveFilterDimensionOption(ctx context.Context, filterID, dime
 	return filter.ETag, nil
 }
 
-// findDimensionIndex loops through a dimension, looking for a filter by name, and
+// findDimension loops through a dimension, looking for a dimension by name, and
+// returns Dimension.
+func findDimension(filter *model.Filter, dimensionName string) (model.Dimension, error) {
+	for _, dim := range filter.Dimensions {
+		if dim.Name == dimensionName {
+			return dim, nil
+		}
+	}
+
+	return model.Dimension{}, errors.New("could not find dimension")
+}
```

Comment:
I would use the `comma-ok` idiom instead.
But it does not add anything to this... **not a blocker at all**.